source/module_hamilt_lcao/module_hcontainer/hcontainer.h

```diff
@@ -0,0 +1,360 @@
+ *       HR.unfix_R();
+ *     ```
+ * 4. use for-loop to do something with atom-pairs with specific R index
+ *    a. loop R-index first and then loop atom-pairs
+ *     ```
+ *       // HR is a const HContainer object, which has been initialized
+ *       int rx, ry, rz;
+ *       // call size_R_loop() to get number of different R indexes, 
+ *       // be careful, it will cost some time to loop all atom-pairs to gather R indexes
+ *       int size_for_loop_R = HR.size_R_loop();
+ *       for (int iR = 0; iR < size_for_loop_R ; iR++)
+ *       {
+ *           // call loop_R() to get R coordinates (rx, ry, rz)
+ *           HR.loop_R(iR, rx, ry, rz);
+ *           // call fix_R() to save atom-pairs with R index (rx, ry, rz) into tmp_atom_pairs
+ *           HR.fix_R(rx, ry, rz);
+ *           // loop fixed atom-pairs
+ *           for (int i = 0; i < HR.size_atom_pairs(); i++)
+ *           {
+ *              // get pointer of target atom-pair
+ *              double* data_pointer = HR.data(i);
+ *              // or get reference of target atom-pair
+ *              AtomPair<double>& atom_ijR = HR.get_atom_pair(i);
+ *              // do something with atom_ijR or data_pointer
+ *              ...
+ *           }
+ *       }
+ *       // call unfix_R() to clear tmp_atom_pairs
+ *       HR.unfix_R();
+ *     ```
+ *    b. loop atom-pairs first and then loop R-index
+ *     ```
+ *       // HR is a const HContainer object, which has been initialized
+ *       // loop atom-pairs
+ *       for (int i = 0; i < HR.size_atom_pairs(); i++)
+ *       {
+ *           // get reference of target atom-pair
+ *           AtomPair<double>& atom_ij = HR.get_atom_pair(i);
+ *           // loop R-index
+ *           for (int iR = 0; iR < atom_ij.size_R(); iR++)
+ *           {
+ *               const int* r_index = atom_ij.get_R_index(iR);
+ *               auto tmp_matrix = atom_ij.get_HR_values(r_index[0], r_index[1], r_index[2]);              
+ *               // do something with tmp_matrix
+ *               ...
+ *           }
+ *       }
+ *     ```
+ *    c. loop atom-pairs with gamma_only case
+ *     ```
+ *       ...
+ *       HR.fix_gamma();
+ *       // HR is a const HContainer object, which has been initialized and fixed to gamma       
+ *       // loop atom-pairs directly without R index
+ *       for (int i = 0; i < HR.size_atom_pairs(); i++)
+ *       {
+ *           // get data pointer of target atom-pair
+ *           double* data_pointer = HR.get_pointer(i);
+ *           // do something with data_pointer
+ *           ...
+ *       }
+ *     ```
+ * 
+*/
+template <typename T>
+class HContainer
+{
+  public:
+    // Destructor of class HContainer
+    ~HContainer(){};
+
+    // copy constructor
+    HContainer(const HContainer<T>& HR_in);
+
+    // move constructor
+    HContainer(HContainer<T>&& HR_in);
+
+    // simple constructor
+    HContainer();
+
+    // use unitcell to initialize atom_pairs
+    HContainer(const UnitCell& ucell_);
```

Comment:
This constructor is unnecessary because (1) it is only used in unit test, (2) Hcontainer should not depends on ucell.